components/server/src/ome/services/ThumbnailBean.java

```diff
@@ -1129,6 +1129,20 @@ protected void actOnOneGroup(Set<Long> pixelsIds) {
                 _createThumbnail();
             }
             byte[] thumbnail = ioService.getThumbnail(thumbnailMetadata);
+            //Thumbnails are copied to disk retrieved from disk including
```

Comment:
Rewording below:

Thumbnails are always saved to disk and then retrieved when the call stack includes `retrieveThumbnail().`  This includes the "clock".  `inProgress` is not set early enough for `retrieveThumbnailDirect()` to be called.

The current implementation assumes that if an image has valid rendering settings for the current user then it will have a valid pyramid.  When existing pyramids, thumbnail metadata, and cached thumbnails are deleted from disk using a method like removepyramids it invalidates this assumption.

The "clock" will first be saved to disk.  `inProgress` will then be checked and the "clock" deleted from disk after creation.  The "clock" will have to be created, saved to disk, and then retrieved every time the thumbnail is requested until pyramid creation completes and the correct thumbnail can be created.  Otherwise the "clock" would stay as the cached thumbnail until the valid rendering settings for the current user are modified.